File: January 2023/19 January 2023/Carpet into Box.java
//{ Driver Code Starts
import java.util.*;
import java.io.*;
import java.lang.*;

class Geeks
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            int A =sc.nextInt();
            int B =sc.nextInt();
            int C =sc.nextInt();
            int D =sc.nextInt();
             
           System.out.println(new Solution().carpetBox(A,B,C,D)); 
        }
    }
}

// } Driver Code Ends


//User function Template for Java


class Solution{
    
    int find(int a,int b)
    {
        int ans=0;
        while(a>b)
        {
            ans++;
            a=(a/2);
        }
        return ans;
    }
    
    int carpetBox(int A, int B, int C, int D) { 
        
        return Math.min(find(A,C)+find(B,D),find(A,D)+find(B,C));
    }
   
}

File: January 2023/19 January 2023/Carpet into Box.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;


// } Driver Code Ends

//User function Template for C++

class Solution{
    public:
    int carpetBox(int A, int B, int C, int D){
        if(max(A,B)<max(C,D) && min(A,B)<min(C,D)){
            return 0;
        }
        else{
            int count =0;
            while(max(A,B)>max(C,D)){
                count++;
                if(A>B){
                    A=floor(A/2);
                }
                else{
                    B=floor(B/2);
                }
            }
            while(min(A,B)>min(C,D)){
                count++;
                if(A>B){
                    B=floor(B/2);
                }
                else{
                    A=floor(A/2);
                }
            }
            return count;
        }
    }
};

//{ Driver Code Starts.


int main(){
    int t;
    cin>>t;
    while(t--){
        int A,B,C,D;
        cin>>A>>B>>C>>D;

        Solution ob;

        int ans = ob.carpetBox(A,B,C,D);

        cout<<ans<<endl;
    }
}
// } Driver Code Ends
